packages/firebase_crashlytics/firebase_crashlytics/lib/src/firebase_crashlytics.dart

```diff
@@ -75,67 +75,72 @@ class FirebaseCrashlytics extends FirebasePluginPlatform {
   }
 
   /// Submits a Crashlytics report of a caught error.
-  Future<void> recordError(dynamic exception, StackTrace stack,
+  Future<void> recordError(dynamic exception, StackTrace? stack,
       {dynamic reason,
-      Iterable<DiagnosticsNode> information,
-      bool /*?*/ printDetails}) async {
-    // If [null] is provided, use the debug flag instead.
-    printDetails ??= kDebugMode;
+      Iterable<DiagnosticsNode> information = const [],
+      bool printDetails = false}) async {
+    // Use the debug flag if provided, otherwise user can set using printDetails
+    bool printOut = kDebugMode || printDetails;
```

Comment:
That's not the same behaviour

Before: null = "is in debug mode"
After: "debug mode always prints"